app/src/main/kotlin/com/wire/android/ui/authentication/login/sso/LoginSSOViewModel.kt

```diff
@@ -90,10 +91,48 @@ class LoginSSOViewModel @Inject constructor(
     }
 
     fun onCustomServerDialogConfirm() {
-        if (loginState.customServerDialogState != null) {
-            authServerConfigProvider.updateAuthServer(loginState.customServerDialogState!!.serverLinks)
+        viewModelScope.launch {
+            if (loginState.customServerDialogState != null) {
+                authServerConfigProvider.updateAuthServer(loginState.customServerDialogState!!.serverLinks)
+
+                val authScope = coreLogic.versionedAuthenticationScope(loginState.customServerDialogState!!.serverLinks)().let {
+                    when (it) {
+                        is AutoVersionAuthScopeUseCase.Result.Failure.Generic,
+                        AutoVersionAuthScopeUseCase.Result.Failure.TooNewVersion,
+                        AutoVersionAuthScopeUseCase.Result.Failure.UnknownServerVersion -> {
+                            loginState = loginState.copy(customServerDialogState = null)
+                            return@launch
+                        }
+
+                        is AutoVersionAuthScopeUseCase.Result.Success -> {
+                            it.authenticationScope
+                        }
+                    }
+                }
+
+                authScope.ssoLoginScope.fetchSSOSettings().also {
+                    when (it) {
+                        is FetchSSOSettingsUseCase.Result.Failure -> {}
+                        is FetchSSOSettingsUseCase.Result.Success -> {
+                            it.defaultSSOCode?.let { ssoCode ->
+                                val ssoCodeWithPrefix = ssoCode.startsWith("wire-").let { hasPrefix ->
+                                    if (hasPrefix) ssoCode else "wire-$ssoCode"
+                                }
```

Comment:
Could it be improved a little bit like this? then we don't need the `.let { .. }` scope 🤔 

```suggestion
                                val ssoCodeWithPrefix = if (ssoCode.startsWith("wire-")) ssoCode else "wire-$ssoCode"
```